src/ngraph/op/fused/clamp.cpp

```diff
@@ -42,13 +135,81 @@ void op::Clamp::pre_validate_and_infer_types()
 NodeVector op::Clamp::decompose_op() const
 {
     const auto data = input_value(0);
-    const auto data_shape = data.get_shape();
+    const auto type = data.get_element_type();
+    const auto shape = data.get_shape();
+
+    auto ceil_func = [](double x) { return std::ceil(x); };
+    auto floor_func = [](double x) { return std::floor(x); };
+
+    shared_ptr<Node> clamp_min, clamp_max;
 
-    const auto clamp_min = builder::make_constant(data.get_element_type(), data_shape, m_min);
-    const auto clamp_max = builder::make_constant(data.get_element_type(), data_shape, m_max);
+    switch (type)
+    {
+    case element::Type_t::i8:
+    {
+        clamp_min = builder::make_constant(type, shape, double_to_int<int8_t>(m_min, ceil_func));
```

Comment:
Don't call make_constant, just do a make_share<Constant> and then you can pass min/max directly as the correct type. There is a ctor for Constant that takes a scalar as the 3rd arg.

Reply:
OK, will fix.

Reply:
Done